apps/client/src/app/[user]/components/RecommendSection/index.tsx

```diff
@@ -1,23 +1,59 @@
+"use client";
+
+import SearchStudyInput from "@/app/[user]/components/SearchStudyInput";
+import { useSearchStudyQueryObject } from "@/app/api/groups/query";
+
+import RecommendList from "@/app/[user]/components/RecommendSection/RecommendList";
+import { useQuery } from "@tanstack/react-query";
+import { useSearchParams } from "next/navigation";
 import RecommendCard from "./RecommendCard";
 import {
+  recommendHeaderContentWrapper,
   recommendHeaderWrapper,
   recommendSectionWrapper,
   recommendStudyTitle,
+  searchedStudyCountStyle,
 } from "./index.css";
 
-const RecommendStudySection = () => (
-  <section
-    className={recommendSectionWrapper}
-    aria-labelledby="recommend-title"
-  >
-    <div className={recommendHeaderWrapper}>
-      <h2 id="recommend-title" className={recommendStudyTitle}>
-        추천 스터디
-      </h2>
-      <input />
-    </div>
-    <RecommendCard />
-  </section>
-);
+const RecommendStudySection = () => {
+  const searchParam = useSearchParams();
+  const searchPattern = searchParam.get("search") || "";
+
+  const { data: studyList } = useQuery(
+    useSearchStudyQueryObject({ searchPattern }),
+  );
+
+  return (
+    <section
+      className={recommendSectionWrapper}
+      aria-labelledby="recommend-title"
+    >
+      <div className={recommendHeaderWrapper}>
+        <div className={recommendHeaderContentWrapper}>
+          <h2 id="recommend-title" className={recommendStudyTitle}>
+            {searchPattern ? "검색 결과" : "추천 스터디"}
+          </h2>
+          {searchPattern && (
+            <div className={searchedStudyCountStyle}>
+              {studyList?.content.length || 0}
+            </div>
+          )}
+        </div>
+
+        <SearchStudyInput />
+      </div>
+
+      {searchPattern ? (
+        <RecommendList studyList={studyList?.content || []} />
```

Comment:
```suggestion
        isFetching ? <Spinner /> : <RecommendList studyList={studyList?.content || []} />
```
useQuery에서 isFetching을 추가로 가져와서 로딩 상태를 구현해주면 좋을 것 같네요. 지금은 검색 fetching이 실행되는 동안 결과가 없다고 떠서요.